internal/service/alarms/internal/db/migrations/000003_create_alarm_event_record_table.up.sql

```diff
@@ -0,0 +1,59 @@
+-- Counter to keep track of the latest events, used to notify only the latest event.
+CREATE SEQUENCE IF NOT EXISTS alarm_sequence_seq
+    START WITH 1
+    INCREMENT BY 1
+    MINVALUE 1
+    NO MAXVALUE
+    CACHE 1;
+
+-- Logs individual alarm events. alarm_event_record also contains all the data needed for generating AlarmEventNotification
+CREATE TABLE IF NOT EXISTS alarm_event_record (
+    -- O-RAN
+    alarm_event_record_id UUID PRIMARY KEY DEFAULT gen_random_uuid(), -- Unique identifier for each event record
+    alarm_definition_id UUID NOT NULL, -- From alarm_definition table
+    probable_cause_id UUID NOT NULL, -- From alarm_definition table
+    alarm_raised_time TIMESTAMPTZ NOT NULL, -- From current alert notification
+    alarm_changed_time TIMESTAMPTZ, -- From current alert notification
+    alarm_cleared_time TIMESTAMPTZ, -- From current alert notification
+    alarm_acknowledged_time TIMESTAMPTZ, -- From current alert notification
+    alarm_acknowledged BOOLEAN NOT NULL DEFAULT FALSE, -- From PATCH api request but default to false
+    perceived_severity INT NOT NULL, -- We will need to map the current alert with this from code
+    extensions JSONB, -- Additional data for extensibility
+
+    -- O-RAN additional data to create AlarmEventNotification
+    resource_id UUID NOT NULL, -- Same as manager_cluster_id for caas alerts
+    resource_type_id UUID NOT NULL, -- Derived from manager_cluster_id
+    notification_event_type INT NOT NULL, -- Should be enum calculated from current alert
+
+    -- Internal
+    alarm_status VARCHAR(20) DEFAULT 'firing' NOT NULL, -- Status of the alarm (either 'firing' or 'resolved'). This is also used to archive it later.
+    finger_print TEXT NOT NULL, -- Unique identifier of caas alerts
```

Comment:
fingerprint is typically spelled as 1 word no spaces or dashes.

Reply:
updated!